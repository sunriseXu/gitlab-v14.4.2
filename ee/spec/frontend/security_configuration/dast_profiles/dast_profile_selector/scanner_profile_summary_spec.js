import { shallowMount } from '@vue/test-utils';
import App from 'ee/security_configuration/dast_profiles/dast_profile_selector/scanner_profile_summary.vue';
import { scannerProfiles } from 'ee_jest/security_configuration/dast_profiles/mocks/mock_data';

const [profile] = scannerProfiles;

describe('DastScannerProfileSummary', () => {
  let wrapper;

  const createWrapper = (props = {}) => {
    wrapper = shallowMount(App, {
      propsData: {
        profile,
        ...props,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders properly', () => {
    createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });
});
