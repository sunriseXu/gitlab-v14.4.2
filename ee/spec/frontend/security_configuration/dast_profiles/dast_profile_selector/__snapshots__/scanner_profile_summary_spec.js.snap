// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DastScannerProfileSummary renders properly 1`] = `
<dast-profile-summary-card-stub
  iseditable="true"
  profiletype="scanner"
>
  
    Passive scanner
  
  <summary-cell-stub
    class=""
    label="Scan mode"
  >
    <scan-type-badge-stub
      scantype="PASSIVE"
    />
  </summary-cell-stub>
   
  <summary-cell-stub
    label="Spider timeout"
    value="5 minutes"
  />
   
  <summary-cell-stub
    label="Target timeout"
    value="10 seconds"
  />
   
  <summary-cell-stub
    label="AJAX spider"
    value="Off"
  />
   
  <summary-cell-stub
    label="Debug messages"
    value="Hide debug messages"
  />
</dast-profile-summary-card-stub>
`;
